modelzoo/ESMM/train.py

```diff
@@ -0,0 +1,720 @@
+    parser.add_argument('--checkpoint_dir',
+                        help='Full path to checkpoints output directory')
+    parser.add_argument('--learning_rate',
+                        help='Learning rate for model',
+                        type=float,
+                        default=0.1)
+    parser.add_argument('--l2_regularization',
+                        help='L2 regularization for the model',
+                        type=float)
+    parser.add_argument('--timeline',
+                        help='number of steps on saving timeline',
+                        type=int)
+    parser.add_argument('--save_steps',
+                        help='set the number of steps on saving checkpoints',
+                        type=int)
+    parser.add_argument('--keep_checkpoint_max',
+                        help='Maximum number of recent checkpoint to keep',
+                        type=int,
+                        default=1)
+    parser.add_argument('--bf16',
+                        help='enable DeepRec BF16 in deep model',
+                        action='store_true')
+    parser.add_argument('--no_eval',
+                        help='not evaluate trained model by eval dataset',
+                        action='store_true')
+    parser.add_argument('--protocol',
+                        type=str,
+                        choices=['grpc'],
+                        default='grpc')
+    parser.add_argument('--inter',
+                        help='set inter op parallelism threads',
+                        type=int,
+                        default=0)
+    parser.add_argument('--intra',
+                        help='set intra op parallelism threads',
+                        type=int,
+                        default=0)
+    parser.add_argument('--input_layer_partitioner',
+                        help='slice size of input layer partitioner. units MB',
+                        type=int,
+                        default=0)
+    parser.add_argument('--dense_layer_partitioner',
+                        help='slice size of dense layer partitioner. units KB',
+                        type=int,
+                        default=0)
+    return parser
+
+if __name__ == '__main__':
+    parser = get_arg_parser()
+    args = parser.parse_args()
+
+    SEED = args.seed
+    tf.set_random_seed(SEED)
+
+    print('Validating arguments')
+    train_file = os.path.join(args.data_location, 'taobao_train_data')
+    test_file = os.path.join(args.data_location, 'taobao_test_data')
+    if not os.path.exists(args.data_location):
+        raise ValueError(f'[ERROR] data location: {args.data_location} does not exist. '
+                         'Please provide valid path')
+
+    if not os.path.exists(train_file) or not os.path.exists(test_file):
+        raise ValueError('[ERROR] taobao_train_data or taobao_test_data does not exist '
+                         'in the given data_location. Please provide valid path')
+
+    if not os.path.exists(args.output_dir):
```

Comment:
Please remove the validation of `args.output_dir`. An error will occur in our regular test for Ali. If this directory does not exist, it will be automatically created.

Reply:
@Duyi-Wang done